### PR TITLE
Add unit test that shows LineBasedFrameDelimiter correctly splits line.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Thought there may be a bug so added a testcase to verify everything works as expected.

Modifications:

Added testcase

Result:

More test-coverage.

### CODE REVIEW

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (7a4b0c3) and 4.0 (f77c6f7)

**Comment by johnou** [comment]
We should probably add another test which covers fragmented packets, I'll create a pull request later today.

**Comment by normanmaurer** [comment]
@johnou yes please... we love PRs ;)

File: codec/src/test/java/io/netty/handler/codec/LineBasedFrameDecoderTest.java [hunk 1]
```diff
@@ -128,4 +128,21 @@ public void testTooLongLineWithFailFast() throws Exception {
...
         buf2.release();
     }
+
+    @Test
+    public void testDecodeSplitsCorrectly() throws Exception {
+        EmbeddedChannel ch = new EmbeddedChannel(new LineBasedFrameDecoder(8192, false, false));
+
+        assertTrue(ch.writeInbound(copiedBuffer("line\r\n.\r\n", CharsetUtil.US_ASCII)));
+
+        ByteBuf buf = ch.readInbound();
```
Comment (johnou) [inline codec/src/test/java/io/netty/handler/codec/LineBasedFrameDecoderTest.java]: missing cast.. (ByteBuf)

**Comment by normanmaurer** [comment]
@johnou ups... let me fix it in 4.0 (works in 4.1)

**Pull request closed without merging**